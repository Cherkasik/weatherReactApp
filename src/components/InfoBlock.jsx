import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import * as weatherSelector from "../weather/selectors";

const AddInfoBox = styled.div`
  width: 97.5%;
  background: #A7AFB1;
  margin-top: 10px;
  padding: 10px;
  color: #FFFFFF;
  display: flex;
  justify-content: space-between;
  
  div:first-child {
    font-size: 14pt;
    font-weight: 700;
  }
  
  div:last-child {
    font-weight: 500;
    font-size: 14pt;
  }
`;


function CurWeather(props) {
  const { weather } = props;
  return (
    <>
      <AddInfoBox>
        <div>Ветер</div>
        <div>{weather.wind.speed} м/с, {weather.wind.deg}°</div>
      </AddInfoBox>
      <AddInfoBox>
        <div>Облачность</div>
        <div>{weather.clouds.all}%</div>
      </AddInfoBox>
      <AddInfoBox>
        <div>Давление</div>
        <div>{weather.main.pressure/1000} Бар</div>
      </AddInfoBox>
      <AddInfoBox>
        <div>Влажность</div>
        <div>{weather.main.humidity}%</div>
      </AddInfoBox>
      <AddInfoBox>
        <div>Координаты</div>
        <div>[{weather.coord.lon}, {weather.coord.lat}]</div>
      </AddInfoBox>
    </>
  );
}

const mapStateToProps = state => ({
  weather: weatherSelector.getWeather(state),
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(CurWeather);
