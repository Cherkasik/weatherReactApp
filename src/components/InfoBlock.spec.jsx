import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import InfoBlock from './InfoBlock';

describe('InfoBlock component', () => {
  const weather = {
    wind: { speed: 10, deg: 10 },
    clouds: { all: 10 },
    main: { pressure: 10000, humidity: 10 },
    coord: { lon: 10, lat: 10 },
  };
  it('should render correctly', () => {
    const { container } = render(<InfoBlock weather={weather} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});