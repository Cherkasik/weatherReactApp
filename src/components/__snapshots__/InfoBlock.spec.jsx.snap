// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBlock component should render correctly 1`] = `
.c0 {
  width: 97.5%;
  background: #A7AFB1;
  margin-top: 10px;
  padding: 10px;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 div:first-child {
  font-size: 14pt;
  font-weight: 700;
}

.c0 div:last-child {
  font-weight: 500;
  font-size: 14pt;
}

<div
  class="c0"
>
  <div>
    Ветер
  </div>
  <div>
    10
     м/с, 
    10
    °
  </div>
</div>
`;
